modules/random_page/random_page.py

```diff
@@ -0,0 +1,99 @@
+                        if not skipindex:
+                            add_to_json = False
+                            # Make sure the page isn't an object index page
+                            if route == "matrices":
+                                add_to_json = True
+                            elif route == "tactics" and re.search(r"TA[0-9]{4}", thepath):
+                                add_to_json = True
+                            elif route == "techniques" and re.search(r"T[0-9]{4}", thepath):
+                                add_to_json = True
+                            elif route == "mitigations" and re.search(r"M[0-9]{4}", thepath):
+                                add_to_json = True
+                            elif route == "groups" and re.search(r"G[0-9]{4}", thepath):
+                                add_to_json = True
+                            elif route == "software" and re.search(r"S[0-9]{4}", thepath):
+                                add_to_json = True
+
+                            if add_to_json:
+                                json_data[value].append(thepath[6:])
+            else:
+                json_data[value] = []
+    
+    if not os.path.isdir(site_config.web_directory):
+        os.makedirs(site_config.web_directory)
+        
+    json.dump(json_data, open(os.path.join(site_config.web_directory, "random_page.json"), mode="w",  encoding="utf-8"), indent=4)
+
+def skipline(line):
+    for skip in skiplines:
+        if skip in line: return True
+    return False
+
+def clean_line(line):
+    """clean unicode spaces from line"""
+    # Replace unicode spaces
+    line = line.replace(u"\u00a0", " ")
+    line = line.replace(u"\u202f", " ")
+    line = line.replace("&nbsp;", " ")
+    line = line.replace("&nbsp", " ")
+
+    return line
+
+def clean(filepath):
+    """clean the file of all HTML tags and unnecessary data"""
+    f = open(filepath, mode="r", encoding="utf8")
+    lines = f.readlines()
+    f.close()
+
+    content = ""
+    title = ""
+    skipindex = False
+    indexing = False
+    for line in lines:
+        if (not skipline(line)) and indexing: 
+            content += clean_line(line) + "\n"
+        if "<!--start-indexing-for-search-->" in line: 
+            indexing = True
+        if "<!--stop-indexing-for-search-->" in line: 
+            indexing = False
+        if "<title>" in line:
+            # e.g [Credential Access - Enterprise | MITRE ATT&CK&trade;] becomes [Credential Access - Enterprise]
+            match = re.search(r"<title>(.*)\|.*</title>", line)
+            if match: title = match.group(1).strip()
+        if 'http-equiv="refresh"' in line: skipindex = True
+
+    out = bleach.clean(content, tags=[], strip=True) #remove tags
+    out = re.sub(r"[\n ]+", " ", out) # remove extra newlines, smush to 1 line
+    out = html.unescape(out) # fix &amp and &#nnn unicode escaping
+
+    skipindex = skipindex or out == "" or out == " "
+    return out, skipindex, title
```

Comment:
90% of this is unnecessary stuff from the search module. We don't need to clean the content, we just need to check if the page has a redirect (line 92) or whether it has nofollow (for deprecated pages, which I don't actually see here. [See this line for how I did it in develop](https://github.com/mitre-attack/attack-website/blob/develop/modules/search.py#L85)).